### PR TITLE
Place Helper module

### PR DESCRIPTION
**Pull request by GenixZero**
This option checks to see if a block is next to the hit result. If enabled, it will loop through the max range to 1 to find the best hit result.

### CODE REVIEW

File: src/main/java/meteordevelopment/meteorclient/systems/modules/world/AirPlace.java [hunk 1]
```diff
@@ -99,4 +105,29 @@ private void onRender(Render3DEvent event) {
 
         event.renderer.box(((BlockHitResult) hitResult).getBlockPos(), sideColor.get(), lineColor.get(), shapeMode.get(), 0);
     }
+
+    private void setHitResult() {
+        final double r = customRange.get() ? range.get() : mc.interactionManager.getReachDistance();
+        for (int i = (int) r; i > 0; i -= 1D) {
+            hitResult = mc.getCameraEntity().raycast(Math.min(r, i), 0, false);
```
Comment (StormyBytes) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/world/AirPlace.java:112]: Why do you need this part though? Doesn't this just ray cast but with decrementing lengths?

Also why do you need to use `Math.min(r, i)` when `i` will always be equal or less than `r`, just use `i` instead.

File: src/main/java/meteordevelopment/meteorclient/systems/modules/world/AirPlace.java [hunk 2]
```diff
@@ -99,4 +105,29 @@ private void onRender(Render3DEvent event) {
...
+        if (!onlyColliding.get()) {
+            return true;
+        }
+        for (Direction dir : Direction.values()) {
+            assert mc.world != null;
+            if (mc.world.getBlockState(pos.add(dir.getOffsetX(), dir.getOffsetY(), dir.getOffsetZ())).getMaterial() != Material.AIR) {
+                return true;
+            }
+        }
+        return false;
```
Comment (StormyBytes) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/world/AirPlace.java:131]: Can't this be simplified to use `BlockUtils.getPlaceSide()` ?

Like,
```
return BlockUtils.getPlaceSide(pos) != null;
```

**Review by StormyBytes — COMMENTED** [review]
Also isn't the whole point of the module named "AirPlace" to *air place* blocks without needing to care whether they have a side to place on or not?

File: src/main/java/meteordevelopment/meteorclient/systems/modules/world/AirPlace.java (hunk 2 above)
Comment (seasnail8169) [reply src/main/java/meteordevelopment/meteorclient/systems/modules/world/AirPlace.java:131]: yes

**Comment by GenixZero** [comment]
Yea, forgot to remove the Math.min when I was testing something else. Also I don't know if BlockUtils.getPlaceSide works, the current method checks to see if a block is next to it if not, then hit result gets set to null. 
This works well for anticheats that check if a block that is placed is next to anything else.
The reason I loop through the range is to get the best placement. If you look straight down to the void while a block is under your feet, then the for loop will return the hit result of the block next to the one you're standing on.

**Comment by StormyBytes** [comment]
Ah my bad, only now do I realize why the raycast needs to be in decrementing lengths. 

Though my point about the air placing still stands. In my opinion this would be better off as a sort of "place helper" module instead of being in the air place module as it just goes against the module's name.

**Comment by GenixZero** [comment]
I agree with you, I made it into its own module.

**Comment by seasnail8169** [comment]
i don't like this, it being in airplace would be better imo, you should just have a couple modes.

mode 1: *only* place if there is a block to place against
mode 2: try to find a side, if not then just place in the air

i will probably just write this myself tho tbh

**Comment by StormyBytes** [comment]
> i will probably just write this myself tho tbh

10 months ago :trollface:

**Comment by I-Got-You-Dead** [comment]
💀

**Pull request closed without merging**